src/NHibernate/Dialect/SQLiteDialect.cs

```diff
@@ -484,17 +498,5 @@ public override bool SupportsForeignKeyConstraintInAlterTable
 		// Said to be unlimited. http://sqlite.1065341.n5.nabble.com/Max-limits-on-the-following-td37859.html
 		/// <inheritdoc />
 		public override int MaxAliasLength => 128;
-
-		[Serializable]
-		protected class SQLiteCastFunction : CastFunction
```

Comment:
It seems this need to be obsoleted as it was `protected` for a while.

Reply:
Your are right, I have been a bit quick on this one.